src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -203,24 +201,26 @@ private static void analyzeRepos(String outputPath, List<RepoConfiguration> conf
             } catch (NoAuthorsWithCommitsFoundException nafe) {
                 logger.log(Level.WARNING, String.format(MESSAGE_NO_AUTHORS_WITH_COMMITS_FOUND,
                         configToAnalyze.getLocation(), configToAnalyze.getBranch()));
-                generateEmptyRepoReport(repoReportDirectory.toString(), Author.NAME_NO_AUTHOR_WITH_COMMITS_FOUND);
+                generatedFiles.addAll(
+                    generateEmptyRepoReport(repoReportDirectory.toString(), Author.NAME_NO_AUTHOR_WITH_COMMITS_FOUND));
             }
         }
+        return generatedFiles;
     }
 
     /**
      * Analyzes repo specified by {@code config} and generates the report.
      */
-    private static void analyzeRepo(
+    private static List<Path> analyzeRepo(
             RepoConfiguration config, String repoReportDirectory) throws NoAuthorsWithCommitsFoundException {
         // preprocess the config and repo
         updateRepoConfig(config);
         updateAuthorList(config);
 
         CommitContributionSummary commitSummary = CommitsReporter.generateCommitSummary(config);
         AuthorshipSummary authorshipSummary = AuthorshipReporter.generateAuthorshipSummary(config);
-        generateIndividualRepoReport(repoReportDirectory, commitSummary, authorshipSummary);
         logger.info(String.format(MESSAGE_COMPLETE_ANALYSIS, config.getLocation(), config.getBranch()));
+        return generateIndividualRepoReport(repoReportDirectory, commitSummary, authorshipSummary);
```

Comment:
The logging message needs to come after the individual reports have been written successfully.

There's no point if the log message comes before successfully generating the json file because users won't even be able to view them if there's issues writing into json file in the first place.

Reply:
Done. I've placed the logging after the report generation function call.